core/src/main/java/com/kin/ecosystem/core/data/settings/SettingsDataSourceLocal.java

```diff
@@ -8,21 +8,19 @@
 
 	private static final String SETTINGS_PREF_NAME_FILE_KEY = "kinecosystem_settings_data_source";
 
-	private static final String IS_BACKED_UP_KEY = "backed_up";
-
 	private final SharedPreferences settingsDataSourceSharedPref;
 
 	public SettingsDataSourceLocal(@NonNull final Context context) {
 		this.settingsDataSourceSharedPref = context.getSharedPreferences(SETTINGS_PREF_NAME_FILE_KEY, Context.MODE_PRIVATE);
 	}
 
 	@Override
-	public void setIsBackedUp(boolean isBackedUp) {
-		settingsDataSourceSharedPref.edit().putBoolean(IS_BACKED_UP_KEY, isBackedUp).apply();
+	public void setIsBackedUp(final String publicAddress, boolean isBackedUp) {
+		settingsDataSourceSharedPref.edit().putBoolean(publicAddress, isBackedUp).apply();
```

Comment:
In the future, we might want to save additional information in setting related to single public address, maybe keep  IS_BACKED_UP_KEY and append publicAddress to it as the key?

Reply:
👍